### PR TITLE
FUSETOOLS-3046 - some refactoring clean-up

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
Did some of the refactoring suggested by @apupier a while back in https://github.com/bfitzpat/fuseide/pull/38. Aurelien, please let me know if there was anything else you were looking for here?

**Comment by bfitzpat** [comment]
Wow there are some fun errors on Jenkins for this one.

**Comment by apupier** [comment]
> Wow there are some fun errors on Jenkins for this one.

general JBoss Tools issue: https://issues.jboss.org/browse/JBIDE-26283

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/actions/DeleteRestConfigurationAction.java [hunk 1]
```diff
@@ -43,8 +47,24 @@ public String getToolTipText() {
...
+		parent.reload();
+	}	
+	
 	@Override
-	public ImageDescriptor getImageDescriptor() {
-		return mImageRegistry.getDescriptor(RestConfigConstants.IMG_DESC_DELETE);
-	}
+	public String getImageName() {
+		return RestConfigConstants.IMG_DESC_DELETE;
+	}
```
[the remark this answers is not in the document]
Comment (bfitzpat) [reply editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/actions/DeleteRestConfigurationAction.java]: Ok. Will look into that. Thanks!

**Comment by bfitzpat** [comment]
Additional clean-up done, but builds still failing due to the upstream issue. All Rest editor tests pass locally with the refactoring at least. :)

**Comment by bfitzpat** [comment]
Should be ready for merge after a review.

**Pull request merged**